src/test/util_tests.cpp

```diff
@@ -603,4 +607,71 @@ BOOST_AUTO_TEST_CASE(test_ParseFixedPoint)
     BOOST_CHECK(!ParseFixedPoint("1.", 8, &amount));
 }
 
+static const std::string LOCKNAME = ".lock";
+
+static void TestOtherThread(fs::path dirname, bool *result)
```

Comment:
Maybe define TestOtherThread as a lambda instead of an external function. TestOther code would seem easier to understand in the context where it launches instead of out here. Also this would allow test case to be self contained and not need external LOCKNAME/TestOther declarations.

Reply:
Conceptually I do prefer (especially) the TestOtherProcess code to be in a self-contained function instead of in-line, because the stuff runs in a separate process. Also making this work on windows (not going to do so in this pull) would involve some factoring in that direction anyhow.
If the LOCKNAME constant is a problem it could be passed in as parameter.

Reply:
But yeah I do see the argument for making it self-contained, too. Really not sure here...